Sources/Build/Command.link(SwiftModule).swift

```diff
@@ -26,7 +26,16 @@ extension Command {
             return [module] + module.recursiveDependencies.filter{ $0.type == .library }
         }.flatMap{ $0 as? SwiftModule }).contents
         
-        var objects = buildables.flatMap { SwiftcTool(module: $0, prefix: prefix, otherArgs: [], executable: linkerExec.asString, conf: conf).objects }
+        var objects = buildables.flatMap { module -> [AbsolutePath] in
+            let tool = SwiftcTool(module: module, prefix: prefix, otherArgs: [], executable: linkerExec.asString, conf: conf)
+            var objects = tool.objects
+            // To make executables debuggable, we need to link swiftmodule of the main executable module.
+            if module.type == .executable && conf == .debug {
+                assert(product.type == .Executable)
+                objects.append(tool.moduleOutputPath)
```

Comment:
As long as the swiftmodule files on disk don't move, just using the main one should be sufficient. If you really wanted to include everything the user built, you'd have to get all of them into the linked binary, which the Swift driver doesn't currently support. (Anders found the right bug discussing the current issues.)